drivers/SmartThings/zigbee-bed/src/shus-mattress/init.lua

```diff
@@ -0,0 +1,422 @@
+    custom_clusters.shus_smart_mattress.attributes.right_back
+  )
+end
+
+local function right_control_waist_cap_handler(driver, device, cmd)
+  process_control_cap(
+    device,
+    cmd.args.waistControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.right_waist
+  )
+end
+
+local function right_control_hip_cap_handler(driver, device, cmd)
+  process_control_cap(
+    device,
+    cmd.args.hipControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.right_hip
+  )
+end
+
+local function yoga_cap_handler(driver, device, cmd)
+  if cmd.args.stateControl == "stop" then
+    device:send(
+      cluster_base.write_manufacturer_specific_attribute(
+        device,
+        custom_clusters.shus_smart_mattress.id,
+        custom_clusters.shus_smart_mattress.attributes.yoga.id,
+        custom_clusters.shus_smart_mattress.mfg_specific_code,
+        custom_clusters.shus_smart_mattress.attributes.yoga.value_type,
+        custom_clusters.shus_smart_mattress.attributes.yoga.value.stop
+      )
+    )
+  elseif cmd.args.stateControl == "left" then
+    device:send(
+      cluster_base.write_manufacturer_specific_attribute(
+        device,
+        custom_clusters.shus_smart_mattress.id,
+        custom_clusters.shus_smart_mattress.attributes.yoga.id,
+        custom_clusters.shus_smart_mattress.mfg_specific_code,
+        custom_clusters.shus_smart_mattress.attributes.yoga.value_type,
+        custom_clusters.shus_smart_mattress.attributes.yoga.value.left
+      )
+    )
+  elseif cmd.args.stateControl == "right" then
+    device:send(
+      cluster_base.write_manufacturer_specific_attribute(
+        device,
+        custom_clusters.shus_smart_mattress.id,
+        custom_clusters.shus_smart_mattress.attributes.yoga.id,
+        custom_clusters.shus_smart_mattress.mfg_specific_code,
+        custom_clusters.shus_smart_mattress.attributes.yoga.value_type,
+        custom_clusters.shus_smart_mattress.attributes.yoga.value.right
+      )
+    )
+  elseif cmd.args.stateControl == "both" then
+    device:send(
+      cluster_base.write_manufacturer_specific_attribute(
+        device,
+        custom_clusters.shus_smart_mattress.id,
+        custom_clusters.shus_smart_mattress.attributes.yoga.id,
+        custom_clusters.shus_smart_mattress.mfg_specific_code,
+        custom_clusters.shus_smart_mattress.attributes.yoga.value_type,
+        custom_clusters.shus_smart_mattress.attributes.yoga.value.both
+      )
+    )
+  end
+end
+
+-- #############################
+-- # Lifecycle handlers define #
+-- #############################
+
+local function device_init(driver, device)
+  -- TODO
+end
+
+local function device_added(driver, device)
+   device:emit_event(custom_capabilities.yoga.supportedYogaState({"stop", "left", "right"}))
+   do_refresh(driver, device)
+end
+
+local function do_configure(driver, device)
+  -- TODO
+end
```

Comment:
these two functions marked `TODO` can be removed

Reply:
Code has been updated according to review recommendations